draft-ietf-mls-protocol.md

```diff
@@ -1287,10 +1287,13 @@ state as follows:
   the null optional value
 * Update the ratchet tree by replacing nodes in the direct
   path from the removed leaf using the information in the Remove message
-* Reduce the size of the roster and the tree until the rightmost
-  element roster element and leaf node are non-null
 * Update the ratchet tree by setting to blank all nodes in the
-  direct path of the removed leaf
+  direct path of the removed leaf, and also setting the root node
+  to blank
+* Truncate the roster such that the last roster element is
+  non-null (clearing the roster if no such element exists)
+* Truncate the tree such that the rightmost non-blank leaf is the
+  last node of the tree (clearing the tree if no such leaf exists)
```

Comment:
I would prefer not to have the "(clearing..." parts.  There's no such thing as a zero-member group, so it would be better to simply disallow Remove on a one-member group.

Reply:
Though now that I say that, I realize that requirement is not totally trivial to enforce, since the last remaining member could have any index.  But it still seems like the most semantically clear approach.

Reply:
What's the issue exactly with a zero-member group? You can't semantically do anything with it, so you just get a useless object at the end of the `Remove`, and you have no choice but to throw it out.

Reply:
Well, that's the problem -- you can't do anything with it.  So it seems like it shouldn't be a valid state.

Reply:
Ah, I was going to say "bad states are states that are semantically incorrect, and there's nothing incorrect about an empty group" but I found a reason why it's not correct (at least for me): `signer_index` is not `optional` in my GroupState object. If the roster is empty, then there is no valid value for `signer_index`. Since this is necessarily an invalid state, I agree that zero-member groups should be disallowed. I'll make an update to reflect this prohibition for Removes.